src/server/storage.zig

```diff
@@ -42,41 +46,43 @@ pub const RegionFile = struct { // MARK: RegionFile
 			return self;
 		};
 		defer main.stackAllocator.free(data);
-		if(data.len < headerSize) {
-			std.log.err("Region file {s} is too small", .{path});
-			return self;
-		}
-		var i: usize = 0;
-		const fileVersion = std.mem.readInt(u32, data[i..][0..4], .big);
-		i += 4;
-		const fileSize = std.mem.readInt(u32, data[i..][0..4], .big);
-		i += 4;
+		self.load(data) catch {
+			std.log.err("Corrupted region file: {s}", .{path});
+		};
+		return self;
+	}
+
+	fn load(self: *RegionFile, data: []const u8) !void {
+		var reader = BinaryReader.init(data, .big);
+
+		const fileVersion = try reader.readInt(u32);
+		const fileSize = try reader.readInt(u32);
+
 		if(fileVersion != version) {
-			std.log.err("Region file {s} has incorrect version {}. Requires version {}.", .{path, fileVersion, version});
-			return self;
+			return error.corrupted;
 		}
-		var sizes: [regionVolume]u32 = undefined;
+
+		var chunkSizes: [regionVolume]u32 = undefined;
 		var totalSize: usize = 0;
-		for(0..regionVolume) |j| {
-			const size = std.mem.readInt(u32, data[i..][0..4], .big);
-			i += 4;
-			sizes[j] = size;
+		for(0..regionVolume) |i| {
+			const size = try reader.readInt(u32);
+			chunkSizes[i] = size;
 			totalSize += size;
 		}
-		std.debug.assert(i == headerSize);
-		if(fileSize != data.len - i or totalSize != fileSize) {
-			std.log.err("Region file {s} is corrupted", .{path});
+
+		if(fileSize != reader.remaining.len or totalSize != fileSize) {
+			return error.corrupted;
 		}
+
 		for(0..regionVolume) |j| {
-			const size = sizes[j];
-			if(size != 0) {
-				self.chunks[j] = main.globalAllocator.alloc(u8, size);
-				@memcpy(self.chunks[j], data[i..][0..size]);
-				i += size;
+			const chunkSize = chunkSizes[j];
```

Comment:
This name is actually more confusing than just `size`, since `chunk.chunkSize` is the size of a chunk in blocks.